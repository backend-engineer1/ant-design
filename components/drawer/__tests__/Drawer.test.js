import React from 'react';
import { mount } from 'enzyme';
import Drawer from '..';

jest.mock('rc-util/lib/Portal');

class DrawerTester extends React.Component {
  constructor(props) {
    super(props);
    this.state = { visible: false };
  }
  componentDidMount() {
    this.setState({ visible: true }); // eslint-disable-line react/no-did-mount-set-state
  }
  saveContainer = (container) => {
    this.container = container;
  }
  getContainer = () => {
    return 'DrawerTester';
  }
  render() {
    return (
      <div>
        <div id="DrawerTester" />
        <Drawer
          {...this.props}
          visible={this.state.visible}
          getContainer={this.getContainer}
        >
          Here is content of Drawer
        </Drawer>
      </div>
    );
  }
}

describe('Drawer', () => {
  it('render correctly', () => {
    const wrapper = mount(<DrawerTester />);
    expect(wrapper.render()).toMatchSnapshot();
  });
});
