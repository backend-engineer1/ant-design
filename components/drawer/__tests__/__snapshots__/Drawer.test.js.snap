// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer closable is false 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right ant-drawer-open"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          style="overflow: auto; height: 100%; width: 256px;"
        >
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer destroyOnClose is true 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
    >
      <div
        class="ant-drawer-content"
      />
    </div>
  </div>
</div>
`;

exports[`Drawer have a title 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right ant-drawer-open"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          style="overflow: auto; height: 100%; width: 256px;"
        >
          <div
            class="ant-drawer-header"
          >
            <div
              class="ant-drawer-title"
            >
              Test Title
            </div>
          </div>
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            />
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`Drawer render correctly 1`] = `
<div
  class=""
>
  <div
    class="ant-drawer ant-drawer-right ant-drawer-open"
  >
    <div
      class="ant-drawer-mask"
    />
    <div
      class="ant-drawer-content-wrapper"
    >
      <div
        class="ant-drawer-content"
      >
        <div
          style="overflow: auto; height: 100%; width: 400px;"
        >
          <button
            aria-label="Close"
            class="ant-drawer-close"
          >
            <span
              class="ant-drawer-close-x"
            />
          </button>
          <div
            class="ant-drawer-body"
          >
            Here is content of Drawer
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
